tools/explorer/pkg/escrow/escrow.go

```diff
@@ -2,491 +2,52 @@ package escrow
-		return errors.Wrap(err, "could not refund escrow")
-	}
-	rpi.Canceled = true
-	if err = types.ReservationPaymentInfoUpdate(e.ctx, e.db, rpi); err != nil {
-		return errors.Wrapf(err, "could not mark escrows for %d as canceled", rpi.ReservationID)
-	}
-	log.Debug().Int64("id", int64(rpi.ReservationID)).Msg("refunded clients for reservation")
+// Run implements the escrow interface
+func (e *Free) Run(ctx context.Context) error {
 	return nil
 }
 
-// payoutFarmers pays out the farmer for a processed reservation
-func (e *Escrow) payoutFarmers(id schema.ID) error {
-	rpi, err := types.ReservationPaymentInfoGet(e.ctx, e.db, id)
-	if err != nil {
-		return errors.Wrap(err, "failed to get reservation escrow info")
-	}
-	if rpi.Released || rpi.Canceled {
-		// already paid
-		return nil
-	}
+// RegisterReservation implements the escrow interface
+func (e *Free) RegisterReservation(reservation workloads.Reservation) (detail types.CustomerEscrowInformation, err error) {
 
-	// collect the farmer addresses and amount they should receive, we already
-	// have sufficient balance on the escrow to cover this
-	paymentInfo := make([]stellar.PayoutInfo, 0, len(rpi.Infos))
-	for _, escrowDetails := range rpi.Infos {
-		// in case of an error in this flow we continue, so we try to pay as much
-		// farmers as possible even if one fails
-		farm, err := e.farmAPI.GetByID(e.ctx, e.db, int64(escrowDetails.FarmerID))
-		if err != nil {
-			log.Error().Msgf("failed to load farm info: %s", err)
-			continue
+	if reservation.NextAction == workloads.NextActionPay {
+		if err = workloadstypes.ReservationSetNextAction(context.Background(), e.db, reservation.ID, workloads.NextActionDeploy); err != nil {
+			err = errors.Wrapf(err, "failed to change state of reservation %d to DEPLOY", reservation.ID)
+			return
 		}
-
-		destination, err := addressByAsset(farm.WalletAddresses, config.Config.Asset)
-		if err != nil {
-			// FIXME: this is probably not ok, what do we do in this case ?
-			log.Error().Err(err).Msgf("failed to find address for %s for farmer %d", config.Config.Asset, farm.ID)
-			continue
-		}
-
-		paymentInfo = append(paymentInfo,
-			stellar.PayoutInfo{
-				Address: destination,
-				Amount:  escrowDetails.TotalAmount,
-			},
-		)
-	}
-
-	addressInfo, err := types.CustomerAddressByAddress(e.ctx, e.db, rpi.Address)
-	if err != nil {
-		log.Error().Msgf("failed to load escrow address info: %s", err)
-		return errors.Wrap(err, "could not load escrow address info")
-	}
-	kp, err := e.wallet.KeyPairFromSeed(addressInfo.Secret)
-	if err != nil {
-		log.Error().Msgf("failed to parse escrow address secret: %s", err)
-		return errors.Wrap(err, "could not load escrow address info")
-	}
-	if err = e.wallet.PayoutFarmers(*kp, paymentInfo, id); err != nil {
-		log.Error().Msgf("failed to pay farmer: %s for reservation %d", err, id)
-		return errors.Wrap(err, "could not pay farmer")
+		reservation.NextAction = workloads.NextActionDeploy
```

Comment:
Reservation is not a pointer, and the object is not returned (or used anywhere else in this flow), so I don't think this will do anything